submissions/MarharytaBoichenko/oop/index.js

```diff
@@ -0,0 +1,78 @@
+class Inhabitant {
+  constructor(name, species, friends, saying) {
+    this.name = name;
+    this.species = species;
+    this.friends = friends;
+    this.saying = saying;
+    this.properties = ["name", "species", "friends", "saying"];
+  }
+  toPrint() {
+    return this.properties
+      .map((prop) => {
+        const neededProp = this[prop] === undefined ? `no ${prop}` : this[prop];
+        return `my ${prop}: ` + neededProp;
+      })
+      .join(", ");
+  }
+}
+
+class Person extends Inhabitant {
+  constructor(name, gender, friends, saying) {
+    super(name, "human", friends, saying);
+    this.gender = gender;
+    this.hands = 2;
+    this.legs = 2;
+    this.properties = [...this.properties, "gender", "hands", "legs"];
+  }
+}
+
+class Man extends Person {
+  constructor(name, friends, saying) {
+    super(name, "man", friends, saying);
+  }
+}
+
+class Woman extends Person {
+  constructor(name, friends, saying) {
+    super(name, "woman", friends, saying);
+  }
+}
+class Animal extends Inhabitant {
+  constructor(name, species, legs, gender, friends, saying) {
+    super(name, species, friends, saying);
+    this.legs = legs;
+    this.gender = gender;
+    this.properties = [...this.properties, "gender", "legs"];
```

Comment:
A child class shouldn't directly change inherited property.
A base class shouldn't handle child's properties.
Altogether, every class should handle end-to-end the properties it owns immediately.
Use method overloading and explicit call to inherited method version.

Reply:
Still an issue.
Use method overloading/overriding and `super` API to call inherited method version. `super` has other useful applications, not only to call parent's class constructor.